src/interfaces.d.ts

```diff
@@ -77,9 +79,28 @@ export interface HasValue {
 
 export interface ItemProperties<T = any> {
 	id: string;
+	index: number;
 	data: T;
 }
 
+/**
+ * @type SizeDetails
+ *
+ * Adds information about the size of a data set
+ *
+ * @property  dataLength  The number of items in the data set available through a slice operation
+ * @property  totalLength The total number of items in the data
+ */
+export interface SizeDetails {
+	dataLength: number;
+	totalLength: number;
```

Comment:
They'd always be known - the data provider will be responsible for managing them. Things like loading rows, etc. are managed by the data provider as well.